### DISCUSSION TITLE
Load color dynamically

### DISCUSSION THREAD

**Discussion by Arumugamsix** [google/model-viewer#4206] [Q&A]
How to Load or apply the material dynamically on page load.

Color path fetch dynamically from URL query params

i have tried this but not working

let createAndApplyTexture = async (channel, event) => {
      let texture = await modelViewerTexture.createTexture(modifiedUrlPath);
      material.pbrMetallicRoughness[channel].setTexture(texture);
    }

    document.querySelector('model-viewer#lazy-load').createTexture('load', (event) => {
      createAndApplyTexture ('baseColorTexture', event);
     // console.log(event.target.value)

    });

**Comment by elalish** [comment]
We'll need a repro link to be able to help.

**Reply by Arumugamsix** [reply]
please find the repo link
https://github.com/Arumugamsix/dynamic-color.git

**Reply by elalish** [reply]
Sorry, I don't mean a git repo - I'm not going to build and serve your code. I mean a Glitch or JSbin or similar - an actual website where I can see the repro on the page.

**Reply by Arumugamsix** [reply]
please find the link 
https://s3.ap-south-1.amazonaws.com/demo.hnscloud.com/AIS/AIS_AR/AR_Update/index.html?webar/assets/Resources/Upvc/Windows/Sliding%20Window/3%20Panel/models/Upvc_Sliding_Window_3T3P&Anthracite%20Grey

**Reply by elalish** [reply]
Nothing 3D is loading on that page, due to 403 Forbidden errors.
